chatkit-android/src/main/kotlin/com/pusher/chatkit/BeamsPushNotifications.kt

```diff
@@ -38,7 +32,25 @@ class BeamsPushNotifications(
     override fun setUserId(userId: String): Result<Unit, Error> {
         val f = FutureValue<Result<Unit, Error>>()
         try {
-            Beams.setUserId(userId, object : Callback<Void, PusherCallbackError> {
+            val tokenProvider = object : TokenProvider {
+                override fun fetchToken(userId: String): String {
+                    val token = beamsTokenProviderService.fetchToken(userId).map { it.token }
+                    when(token) {
+                        is Result.Success -> {
+                            val tokenString = token.value
+                            if (tokenString == null) {
+                                throw Exception("Could not get auth token for push notification service")
+                            }
+                            return tokenString
+                        }
+                        is Result.Failure -> {
+                            throw Exception("Could not authenticate with push notifications service: ${token.error}")
+                        }
+                    }
+                }
```

Comment:
This method doesn't compile for me (there's no return, but it's a curly brace block)

I've made a PR to this with a slightly different implementation (pushing this marshalling of the response up to the service is comes from so the interfaces match more naturally)